libsolidity/ast/ASTPrinter.cpp

```diff
@@ -579,8 +582,11 @@ void ASTPrinter::printSourcePart(ASTNode const& _node)
 	if (!m_source.empty())
 	{
 		SourceLocation const& location(_node.location());
-		*m_ostream << indentation() << "   Source: "
-				   << escaped(m_source.substr(location.start, location.end - location.start), false) << endl;
+		*m_ostream <<
+			indentation() <<
+			"   Source: " <<
+			jsonCompactPrint(m_source.substr(location.start, location.end - location.start)) <<
```

Comment:
See https://github.com/ethereum/solidity/pull/2801#discussion_r135225060

Reply:
```
JSONCPP_STRING write(const Value& root) JSONCPP_OVERRIDE;
```

So I guess it creates a copy of the string?

I'd prefer if we could just do it with a simple regex. Is that possible?

Reply:
What is the advantage? I would prefer not to create our own hand-written code if we already have code that does exactly the same thing in a library.